src/core/ext/filters/client_channel/resolver/dns/c_ares/grpc_ares_wrapper.cc

```diff
@@ -96,12 +101,38 @@ static void grpc_ares_request_ref(grpc_ares_request* r) {
   gpr_ref(&r->pending_queries);
 }
 
+static void log_address_sorting_list(grpc_lb_addresses* lb_addrs,
+                                     const char* input_output_str) {
+  for (size_t i = 0; i < lb_addrs->num_addresses; i++) {
+    char* addr_str;
+    if (grpc_sockaddr_to_string(&addr_str, &lb_addrs->addresses[i].address,
+                                true)) {
+      gpr_log(GPR_INFO,
+              "C-ares sockaddr address sorting %s index: %" PRIdPTR
+              ". Sockaddr-to-string: %s",
+              input_output_str, i, addr_str);
+      gpr_free(addr_str);
+    } else {
+      gpr_log(GPR_INFO,
+              "Failed to convert sockaddr c-ares address sorting %s index: "
```

Comment:
Keep it simple:

    "c-ares address_sorting: %s[%zu]=<unprintable>"

Reply:
same issue for `%zu` as above